service/contract_call_data.go

```diff
@@ -173,5 +169,8 @@ func encodeMethodArgsStrings(methodID []byte, methodSig string, methodArgs []str
 // contractCallMethodID calculates the first 4 bytes of the method
 // signature for function call on contract
 func contractCallMethodID(methodSig string) ([]byte, error) {
+	if len(methodSig) == 0 {
```

Comment:
I think you need this to be < 4 based on your access below?

Reply:
This `crypto.Keccak256([]byte(methodSig))[:4]`  is just to take the first 4 bytes of the Keccak-256 hash rather than the first 4 bytes of the method signature string. 

So checking the string not empty should be enough.